core/src/com/unciv/logic/map/MapUnit.kt

```diff
@@ -105,6 +105,9 @@ class MapUnit {
     var promotions = UnitPromotions()
     var due: Boolean = true
     var isTransported: Boolean = false
+    
+    var religionSpreadsUsed: Int = 0
+    var religion: String? = null
```

Comment:
HMMM I wonder about this. Not sure if a sort of hashmap<string,string> of param to value is better. Because I feel we will have similar cases in the future.
Maybe we can start using templates IN THE UNIT NAME? Like "Great Prophet ([religionName])"

Reply:
> Maybe we can start using templates IN THE UNIT NAME?
I have tried to get this working, and eventually gave up. The problem is that the name is used as a key in _many_ places, including the filename of the icon it should use, and there are bugs everywhere. For now I'll keep using this field, and change the displayname instead.

Reply:
I see